### PR TITLE
Add SysV-style init script and systemd service unit configuration

### PR DESCRIPTION
**Pull request by justintime32**
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackhq/go-audit/blob/master/.github/CONTRIBUTING.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://github.com/slackhq/go-audit/blob/master/CODE_OF_CONDUCT.md).
* [x] I've been mindful about doing atomic commits, adding documentation to my changes, not refactoring too much.
* [x] I've a descriptive title and added any useful information for the reviewer. Where appropriate, I've attached a screenshot and/or screencast (gif preferrably).
* [x] I've written tests to cover the new code and functionality included in this PR.
* [x] I've read, agree to, and signed the [Contributor License Agreement (CLA)](https://docs.google.com/a/slack-corp.com/forms/d/1q_w8rlJG_x_xJOoSUMNl7R35rkpA7N6pUkKhfHHMD9c/viewform).

#### PR Summary

This change adds a SysV-style init script and a systemd service unit configuration to manage go-audit as a service. The SysV init script is useful, in particular, on EL6-based systems (e.g. RHEL 6 and CentOS 6). The systemd service unit configuration can be used on systemd-based systems, including EL7. I have tested these on CentOS 6 and CentOS 7.

### CODE REVIEW

**Comment by nbrownus** [comment]
Thanks for the PR! Since we already have a [systemd script](https://github.com/slackhq/go-audit/blob/master/examples/go-audit/systemd.service) in the examples (which should be moved into `contrib`) could you amend this for just sysv?

**Comment by justintime32** [comment]
Ah, I totally missed that. I'll see if I can chop that commit off of this PR.

Is the contrib folder the appropriate place for this to go?

**Comment by nbrownus** [comment]
You bet, I just forgot to move ours over there.

**Comment by justintime32** [comment]
Alright, the systemd spec file was removed.

Any other thoughts?

File: contrib/go-audit.init [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+CONFIG_FILE="/etc/go-audit.yaml"
+LOG_DIRECTORY="/var/log/go-audit"
+
+[ -e /etc/sysconfig/$prog ] && . /etc/sysconfig/$prog
+
+start() {
+    [ -f $pidfile ] && checkpid $(cat $pidfile) && echo "$prog already running" && return
+
+    echo -n $"Starting $prog: "
+    nohup $exec --config $CONFIG_FILE >>${LOG_DIRECTORY}/go-audit.out 2>>${LOG_DIRECTORY}/go-audit.err </dev/null &
```
Comment (pradeepchhetri) [inline contrib/go-audit.init]: `nohup` blocks SIGHUP which is used while reloading services, so i would like to suggest to use `daemonize` instead.

Comment (justintime32) [inline contrib/go-audit.init]: I don't think go-audit has any such signal handling currently. Sending a SIGHUP would terminate it.

Also, I can't find `daemonize` as an init function. There is a `daemon` function on CentOS 6, but that doesn't actually daemonize the program, it just provides a way of changing the UID and setting a cgroup and a couple other things.

Comment (pradeepchhetri) [inline contrib/go-audit.init]: @justintime32 Ahh, i use CentOS 7. It makes sense. Thanks.

**Comment by justintime32** [comment]
@nbrownus any updates on this PR?

**Comment by nbrownus** [comment]
A quick search looks like `daemon` might be a [better approach](http://unix.stackexchange.com/questions/216024/what-is-the-canonical-way-to-start-a-daemon-in-rhel-centos-6-init-script)

But it's been a long time since I bothered with sysv, sadly, so I'm not much help in regards to the actual init script. If it works for you on your targeted platform then it should be fine. I would just change the name to reflect that, something along the lines of `sysv.centos6.go-audit`.

**Comment by justintime32** [comment]
I considered the `daemon` function but ultimately I didn't think it would provide any value. It doesn't actually daemonize a service, so you still need to run it with nohup and IO redirection. It can change the UID that the process runs as but we're running it as root here so that didn't seem important.

How about naming it `rh-sysv.go-audit.init`? In theory it should work on any RH-based distro still running SysV.

File: contrib/go-audit.init [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    return $retval
+}
+
+restart() {
+    stop
+    start
+}
+
+rh_status() {
+    status -p $pidfile $prog
```
Comment (rcrowley) [inline contrib/go-audit.init]: Isn't `status`(8) part of Upstart?

Comment (justintime32) [inline contrib/go-audit.init]: @rcrowley It's actually an init function provided by `/etc/rc.d/init.d/functions`.

**Pull request merged**